### ISSUE TITLE
Scrub for broken links

### ISSUE THREAD

**Issue by dtolnay** [carllerche/mio#683]
The documentation of mio::event::Event has one: [readiness state]. I recall noticing a few others. Let's go through systematically and clean these up.

---

![selection_060](https://user-images.githubusercontent.com/1940490/29482927-b561716a-844f-11e7-8bf7-5f5bfc79d263.png)

**Comment by dtolnay** [comment]
This can happen in two phases -- it would be a huge help if someone could run through the docs and make a list of the broken links in a comment here, then they can be fixed in bulk. Even going through part of the docs would be helpful. Just let us know which parts you covered.

Bonus points for implementing a warning in rustdoc to catch these automatically!

**Comment by dtolnay** [comment]
[set_readiness] and [portability] are broken in mio::Registration.

**Comment by mayhewj** [comment]
I went through every page on the docs and fixed most of them. The only problem I ran into is that `Events` is exported as both `mio::Events` and `mio::event::Events` and I can only get the links on one of them working due to needing the `../` prefix for the `mio::event::Events` one.

I'm not too familiar with how links work in Rust docs and couldn't find any documentation on this. Any suggestions on how we could handle this?
